sdks/java/extensions/zetasketch/src/test/java/org/apache/beam/sdk/extensions/zetasketch/HllCountTest.java

```diff
@@ -0,0 +1,373 @@
+
+  static {
+    HyperLogLogPlusPlus<Integer> hll = new HyperLogLogPlusPlus.Builder().buildForIntegers();
+    INTS2.forEach(hll::add);
+    INTS2_SKETCH = hll.serializeToByteArray();
+    INTS2_ESTIMATE = hll.longResult();
+  }
+
+  private static final byte[] INTS1_INTS2_SKETCH;
+
+  static {
+    HyperLogLogPlusPlus<?> hll = HyperLogLogPlusPlus.forProto(INTS1_SKETCH);
+    hll.merge(INTS2_SKETCH);
+    INTS1_INTS2_SKETCH = hll.serializeToByteArray();
+  }
+
+  // Long
+  private static final List<Long> LONGS = Collections.singletonList(1L);
+  private static final byte[] LONGS_SKETCH;
+
+  static {
+    HyperLogLogPlusPlus<Long> hll = new HyperLogLogPlusPlus.Builder().buildForLongs();
+    LONGS.forEach(hll::add);
+    LONGS_SKETCH = hll.serializeToByteArray();
+  }
+
+  private static final byte[] LONGS_EMPTY_SKETCH;
+
+  static {
+    HyperLogLogPlusPlus<Long> hll = new HyperLogLogPlusPlus.Builder().buildForLongs();
+    LONGS_EMPTY_SKETCH = hll.serializeToByteArray();
+  }
+
+  // String
+  private static final List<String> STRINGS = Arrays.asList("s1", "s2", "s1", "s2");
+  private static final byte[] STRINGS_SKETCH;
+
+  static {
+    HyperLogLogPlusPlus<String> hll = new HyperLogLogPlusPlus.Builder().buildForStrings();
+    STRINGS.forEach(hll::add);
+    STRINGS_SKETCH = hll.serializeToByteArray();
+  }
+
+  private static final int TEST_PRECISION = 20;
+  private static final byte[] STRINGS_SKETCH_TEST_PRECISION;
+
+  static {
+    HyperLogLogPlusPlus<String> hll =
+        new HyperLogLogPlusPlus.Builder().normalPrecision(TEST_PRECISION).buildForStrings();
+    STRINGS.forEach(hll::add);
+    STRINGS_SKETCH_TEST_PRECISION = hll.serializeToByteArray();
+  }
+
+  // Bytes
+  private static final byte[] BYTES0 = {(byte) 0x1, (byte) 0xa};
+  private static final byte[] BYTES1 = {(byte) 0xf};
+  private static final List<byte[]> BYTES = Arrays.asList(BYTES0, BYTES1);
+  private static final byte[] BYTES_SKETCH;
+
+  static {
+    HyperLogLogPlusPlus<ByteString> hll = new HyperLogLogPlusPlus.Builder().buildForBytes();
+    BYTES.forEach(hll::add);
+    BYTES_SKETCH = hll.serializeToByteArray();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testInitForIntegersGlobally() {
+    PCollection<byte[]> result =
+        p.apply(Create.of(INTS1)).apply(HllCount.Init.forIntegers().globally());
+
+    PAssert.thatSingleton(result).isEqualTo(INTS1_SKETCH);
+    p.run();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testInitForLongsGlobally() {
+    PCollection<byte[]> result =
+        p.apply(Create.of(LONGS)).apply(HllCount.Init.forLongs().globally());
+
+    PAssert.thatSingleton(result).isEqualTo(LONGS_SKETCH);
+    p.run();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testInitForLongsGlobally_EmptyInput() {
+    PCollection<byte[]> result =
+        p.apply(Create.empty(TypeDescriptor.of(Long.class)))
+            .apply(HllCount.Init.forLongs().globally());
```

Comment:
This is unexpected to me, I'd though we'd return an empty accumulator (a serialized HyperLogLogPlusPlus, with no items added to it). Am I missing something about the code in HllCountInitFn, or is this always the case in Beam, that an empty aggregation yields an empty result? Which we could not change in the HllCount code?

Reply:
This behavior is defined in `return Combine.globally(initFn).withoutDefaults();` (L243 in HllCountTest) and documented in `<p>Returns an empty output {@code PCollection} if the input {@code PCollection} is empty.` (L240).

We can change this behavior to return an sketch of an empty set. I made this decision such that it will be consistent with `HllCount.MergePartial`, for which we much return nothing because there is not "identity element" we can return.